includes/components/class-message.php

```diff
@@ -86,6 +89,39 @@ public function __construct( $args = [] ) {
 		parent::__construct( $args );
 	}
 
+    /**
+     * Validates message form.
+     *
+     * @param array  $errors Form errors.
+     * @param object $form Form object.
+     * @return array
+     */
+    public function validate_message_form( $errors, $form ) {
+
+        if ( ! empty( $errors ) ) {
+            return $errors;
+        }
+
+        // Get recipient ID.
+        $recipient_id = $form->get_value( 'recipient' );
+
+        // Get blocked users.
+        $blocked_users = (array) get_user_meta( get_current_user_id(), hp\prefix( 'blocked_users' ), true );
```

Comment:
Avoid using context in callbacks like this one, e.g. the current user is not always the one sending message (it may be created by some third-party API, e.g. from a third-party app and the current user would be set to admin), check if it's possible to fetch the sender ID from the model.

Reply:
I've tried and recipient value from $form->get_model()->get_recipient() is NULL

Reply:
It seems it will be ok to move this checking logic to controller for message_send_action. It is possible to show errors in the form from rest action